cmd/surf/commands/listdocuments.go

```diff
@@ -0,0 +1,64 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+	"github.com/CloudHub360/ch360.go/ch360"
+	"github.com/CloudHub360/ch360.go/config"
+	"gopkg.in/alecthomas/kingpin.v2"
+	"os"
+	"strconv"
+)
+
+type ListDocumentsCmd struct {
+	Client ch360.DocumentGetter
+}
+
+// Configures kingpin with the 'list documents' command
+func ConfigureListDocumentsCmd(ctx context.Context, parentCmd *kingpin.CmdClause, flags *config.GlobalFlags) {
+	listDocumentsCmd := &ListDocumentsCmd{}
+	parentCmd.Command("documents", "List all available documents.").
+		Alias("document").
+		Action(func(parseContext *kingpin.ParseContext) error {
+			err := listDocumentsCmd.initFromArgs(flags)
+			if err != nil {
+				return err
+			}
+			return listDocumentsCmd.Execute(ctx)
+		})
+}
+
+// Executes the command.
+func (cmd *ListDocumentsCmd) Execute(ctx context.Context) error {
+	documents, err := cmd.Client.GetAll(ctx)
+	if err != nil {
+		return err
+	}
+
+	if len(documents) == 0 {
+		fmt.Println("No documents found.")
+		return nil
+	}
+
+	table := NewTable(os.Stdout, []string{"ID", "Size", "Type", "SHA256"})
+
+	for _, document := range documents {
+		table.Append([]string{document.Id, strconv.Itoa(document.Size), document.FileType,
+			document.Sha256})
+	}
+	table.Render()
+
+	return nil
+}
+
+func (cmd *ListDocumentsCmd) initFromArgs(flags *config.GlobalFlags) error {
```

Comment:
💪 Code Climate has flagged this function as duplicated code (also found in listclassifiers.go and listmodules.go). Can it be moved/generalised so it can be reused?

Reply:
Hmm. There were some redundant `var err error` declarations in there (which I've removed), but I can't see any more ways to de-dupe.